lib/src/main/java/uk/gov/di/ipv/cri/address/library/service/JWTVerifier.java

```diff
@@ -31,24 +32,52 @@
 
 public class JWTVerifier {
 
-    public void verifyJWT(
+    public void verifyAuthorizationJWT(
+            Map<String, String> clientAuthenticationConfig, SignedJWT signedJWT)
+            throws SessionValidationException, ClientConfigurationException {
+        verifyJWT(
+                clientAuthenticationConfig,
+                signedJWT,
+                Set.of(
+                        JWTClaimNames.EXPIRATION_TIME,
+                        JWTClaimNames.SUBJECT,
+                        JWTClaimNames.NOT_BEFORE),
+                new JWTClaimsSet.Builder()
+                        .issuer(clientAuthenticationConfig.get("issuer"))
+                        .audience(clientAuthenticationConfig.get("audience"))
+                        .build());
+    }
+
+    public void verifyAccessTokenJWT(
+            Map<String, String> clientAuthenticationConfig, SignedJWT signedJWT, ClientID clientID)
+            throws SessionValidationException, ClientConfigurationException {
+        Set<String> requiredClaims =
+                Set.of(
+                        JWTClaimNames.EXPIRATION_TIME,
+                        JWTClaimNames.SUBJECT,
+                        JWTClaimNames.ISSUER,
+                        JWTClaimNames.AUDIENCE,
+                        JWTClaimNames.JWT_ID);
+        JWTClaimsSet expectedClaimValues =
+                new JWTClaimsSet.Builder()
+                        .issuer(clientID.getValue())
+                        .subject(clientID.getValue())
+                        .audience(clientAuthenticationConfig.get("audience"))
+                        .build();
+        verifyJWT(clientAuthenticationConfig, signedJWT, requiredClaims, expectedClaimValues);
+    }
```

Comment:
this is the main reason for this PR - to indicate the required Claims and expected claim values separately for the Auth Jar JWT verification, and the private_key_jwt verification.

This allows us to assert that client id, iss and sub are the same for the private_key_jwt method.